### PR TITLE
Cleanup main.yml

### PR DESCRIPTION
**Pull request by john-h-kastner**
Some changes that hopefully make the workflow file easier to work with. Mostly just pulls some long paths into environment variables and splits up overly long lines.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -53,7 +55,11 @@ jobs:
       - name: Build 3c
         run: |
           cd ${{env.builddir}}
-          cmake -G Ninja -DLLVM_TARGETS_TO_BUILD=X86 -DLLVM_ENABLE_RTTI=ON -DCMAKE_BUILD_TYPE="Debug" -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra" ${{github.workspace}}/depsfolder/checkedc-clang/llvm
+          cmake -G Ninja\
+            -DLLVM_TARGETS_TO_BUILD=X86\
+            -DCMAKE_BUILD_TYPE="Debug"\
+            -DLLVM_ENABLE_PROJECTS="clang"\
```
Comment (mattmccutchen-cci) [inline .github/workflows/main.yml:61]: I prefer a space before the line continuation character: it makes it easier to see that the character isn't part of the previous argument and makes `git --word-diff` work better.  Please see the `format_and_cleanup.matt` branch.  (I hope I didn't break anything else there; I can run the workflow if you want.)

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -97,32 +106,49 @@ jobs:
           cd ${{env.benchmark_conv_dir}}
           tar -xvzf ${{env.benchmark_tar_dir}}/ptrdist-1.1.tar.gz
           cd ptrdist-1.1
-          for i in anagram bc ft ks yacr2 ; do cd $i ; bear make LOCAL_CFLAGS="-std=c99" ; cd .. ; done
+          for i in anagram bc ft ks yacr2 ; do\
+            (cd $i ; bear make LOCAL_CFLAGS="-std=c99 -Wno-error")\
```
Comment (mattmccutchen-cci) [inline .github/workflows/main.yml:110]: Why did you add `-Wno-error`?

**Review by mattmccutchen-cci — CHANGES_REQUESTED** [review]
Nice cleanup.  I imagine there's more we could do in the future if we changed `convert_project` too.

**Comment by mattmccutchen-cci** [comment]
@john-h-kastner Are you ready for me to re-review this?

**Comment by john-h-kastner** [comment]
> @john-h-kastner Are you ready for me to re-review this?

waiting on successful build

File: .github/workflows/main.yml (hunk 2 above)
Comment (mattmccutchen-cci) [reply .github/workflows/main.yml]: FTR: [This was to make PtrDist pass](https://github.com/correctcomputation/actions/pull/5/commits/745b7a2ce8393ee0dd21c521ee72078d7286d82f).

**Review by mattmccutchen-cci — APPROVED** [review]
